lib/https.js

```diff
@@ -53,25 +53,28 @@ let debug = require('internal/util/debuglog').debuglog('https', (fn) => {
   debug = fn;
 });
 const { URL, urlToHttpOptions, searchParamsSymbol } = require('internal/url');
+const { validateObject } = require('internal/validators');
 
 function Server(opts, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
   if (typeof opts === 'function') {
     requestListener = opts;
-    opts = undefined;
-  }
-  opts = { ...opts };
-
-  if (!opts.ALPNProtocols) {
-    // http/1.0 is not defined as Protocol IDs in IANA
-    // https://www.iana.org/assignments/tls-extensiontype-values
-    //       /tls-extensiontype-values.xhtml#alpn-protocol-ids
-    opts.ALPNProtocols = ['http/1.1'];
+    opts = kEmptyObject;
+  } else if (opts == null) {
+    opts = kEmptyObject;
+  } else {
+    validateObject(opts, 'options');
   }
 
   FunctionPrototypeCall(storeHTTPOptions, this, opts);
-  FunctionPrototypeCall(tls.Server, this, opts, _connectionListener);
+  FunctionPrototypeCall(tls.Server, this,
+                        { ...opts,
```

Comment:
```suggestion
                        { noDelay: true, ...opts,
```
or is there a reason for not include it?

Reply:
I don't have any reason about it. Just I'm not sure passing `noDelay` option in https. Applied your suggestion. Thank you for correction.